operator-pipeline-images/operatorcert/entrypoints/static_tests.py

```diff
@@ -77,13 +80,25 @@ def check_bundle(
     """
     repo = Repo(repo_path)
     operator = repo.operator(operator_name)
-    bundle = operator.bundle(bundle_version)
+    bundle = operator.bundle(bundle_version) if bundle_version else None
+    catalogs = OperatorCatalogList(
+        [
+            catalog
+            for catalog in operator.all_operator_catalogs()
```

Comment:
I am not sure if I understand this loop. The `affected_catalogs.split(",")` returns a list of strings, right? The `operator.all_operator_catalogs()` returns a list of `OperatorCatalog`. And then you compare if `OperatorCatalog` in a list of strings. How it this going to work? Or am I missing anything here?

Also if you already have a list of affected catalogs, why don't you initiate it directly using `OperatorCatalog(<path_to_operator_catalog>)` constructor and instead use an operator and its edge to get to it? If we ever make these types completely independent (operator and operatorCatalog) it won't work.

Reply:
Operator catalog is [comparable to string](https://github.com/mporrato/operator-repo/blob/277471a08a7e9f06169925fcf25b039ac70c4677/src/operator_repo/core.py#L599) and it works both ways. 

I don't understand what you mean by `If we ever make these types completely independent`? Is there such a plan? As of now, we always have an Operator (as it is affected by the catalog change iirc) and using the edge give me all its catalogs. I can see that there is the way to get specific catalog like `OperatorCatalog(repo.root / "catalogs" / "v4.13/kiali")` - for me having the path already provided by the Operator sounds more straightforward, but I am open to discussion.

Reply:
Changed the logic to use `repo.catalog_path()`

Reply:
@Allda I reverted this logic back to use the `operator` to `operator_catalogs` edge - the `affected_catalogs` contain also removed catalogs and this way we can filter them out. Initiate catalog directly `OperatorCatalog(<path_to_operator_catalog>)` raise exception for deleted operator catalogs.